pages/api/applications/[applicationId]/tasks/[taskId].ts

```diff
@@ -1,5 +1,116 @@
+  [MessageType.INVALID_DUE_DATE]: { type: StatusMessageType.ERROR, message: 'The task due date is invalid.' },
+  [MessageType.INVALID_NOTIFICATION_DATETIME]: {
+    type: StatusMessageType.ERROR,
+    message: 'The task notification date is invalid.',
+  },
+  [MessageType.INVALID_IS_DONE]: { type: StatusMessageType.ERROR, message: 'The task completion status is invalid.' },
+});
+
+async function handler(uid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<TaskData>>) {
+  const { taskId, applicationId } = req.query as { taskId: string; applicationId: string };
+
+  if (!Number.isInteger(Number(taskId))) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[MessageType.INVALID_TASK_ID]));
+  }
+
+  if (!Number.isInteger(Number(applicationId))) {
+    return res
+      .status(HttpStatus.BAD_REQUEST)
+      .json(createJsonResponse({}, messages[MessageType.INVALID_APPLICATION_ID]));
+  }
+
+  const task = await prisma.task.findUnique({
+    where: { id: Number(taskId) },
+    include: { application: { select: { id: true, userId: true } } },
+  });
+  if (!task || task.application.id !== Number(applicationId) || task.application.userId !== uid) {
+    return res.status(HttpStatus.NOT_FOUND).json(createJsonResponse({}, messages[MessageType.TASK_NOT_FOUND]));
+  }
+
+  switch (req.method) {
+    case HttpMethod.PATCH:
+      return patchHandler(task.id, req, res);
+    case HttpMethod.DELETE:
+      return deleteHandler(task.id, req, res);
+    default:
+      rejectHttpMethod(res, req.method);
+  }
+}
+
+async function patchHandler(taskId: number, req: NextApiRequest, res: NextApiResponse<ApiResponse<TaskData>>) {
+  const validationError = validateTaskPatchData(req);
+  if (validationError !== null) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[validationError]));
+  }
+
+  const data: TaskPatchData = req.body;
+  const updatedTask: Task = await prisma.task.update({ where: { id: taskId }, data });
+  return res.status(HttpStatus.OK).json(createJsonResponse(updatedTask, messages[MessageType.TASK_PATCHED]));
 }
+
+async function deleteHandler(taskId: number, _req: NextApiRequest, res: NextApiResponse<ApiResponse<TaskData>>) {
+  await prisma.task.delete({ where: { id: taskId } });
+  return res.status(HttpStatus.OK).json(createJsonResponse({}, messages[MessageType.TASK_DELETED]));
+}
+
+function validateTaskPatchData(req: NextApiRequest): MessageType | null {
+  const { title, dueDate, notificationDateTime, isDone } = req.body;
+
+  const isValidTitle = title === undefined || (typeof title === 'string' && title?.trim().length === 0);
+  const isValidDueDate = dueDate === undefined || !isValidDate(dueDate + '');
+  const isValidNotificationDateTime =
+    notificationDateTime === undefined ||
+    notificationDateTime === null ||
+    (typeof notificationDateTime === 'string' && !isValidDate(notificationDateTime));
```

Comment:
```suggestion
    (typeof notificationDateTime === 'string' && isValidDate(notificationDateTime));
```
Condition flipped here as well